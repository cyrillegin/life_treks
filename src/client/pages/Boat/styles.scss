$grey: #e0e0e0;
$shadow1: rgba(172, 172, 172, 0.7);
$shadow2: rgba(76, 76, 76, 0.7);
$shadow3: rgba(0, 0, 0, 0.7);
$green: rgba(0, 197, 0, 1);
$black: #000;
$white: #fff;
$red: rgba(0, 197, 0, 1);

.canvas {
  background: $grey;
  height: 600px;
  margin: 16px auto;
  width: 70%;
}

.cube {
  height: 50px;
  position: absolute;
  right: calc(15% + 33px);
  top: 176px;
  transform: rotateX(55deg) rotateY(0deg) rotateZ(45deg);
  transform-style: preserve-3d;
  transition: transform 1s;
  width: 50px;
}

@mixin base-button() {
  border: 1px solid $black;
  color: $white;
  font-size: 20px;
  font-weight: bold;
  height: 50px;
  line-height: 50px;
  position: absolute;
  text-align: center;
  width: 50px;

  &:hover {
    background: $red;
  }
}

.camera-top {
  @include base-button();
  background: $shadow1;
  transform: rotateY(0deg) translateZ(25px);
}

.camera-side {
  @include base-button();
  background: $shadow2;
  transform: rotateY(90deg) translateZ(25px);
}

.camera-front {
  @include base-button();
  background: $shadow3;
  transform: rotateX(-90deg) translateZ(25px);
}

.camera-perspective {
  background: $shadow1;
  border: 1px solid $black;
  color: $white;
  font-size: 21px;
  font-weight: bold;
  padding: 3px;
  position: absolute;
  right: calc(15% + 38px);
  text-align: center;
  top: 254px;

  &:hover {
    background: $green;
  }
}

.toggles-container {
  left: calc(15% + 20px);
  position: absolute;
  top: 160px;
  width: 20%;
}

.description {
  margin: 16px auto;
  padding: 12px;
  width: 70%;
}
