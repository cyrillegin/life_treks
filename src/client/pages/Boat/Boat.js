import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';

const styles = theme => ({

});

export class BoatPage extends Component {

  static propTypes = {
    classes: PropTypes.object.isRequired,
  }

  render() {
    return (
      <div>
        Hello boat
      </div>
    );
  }
}

export default withStyles(styles)(BoatPage);
