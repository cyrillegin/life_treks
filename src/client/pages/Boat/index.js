import React, { Component } from 'react';
import initScene from './controllers/scene.controller';
import initLights from './controllers/lights.controller';
import CameraController from './controllers/camera.controller';
import MeshController from './controllers/mesh.controller';
import CurvesController from './controllers/curves.controller';
import './styles.scss';

export default class Boat extends Component {
  constructor() {
    super();
    this.state = {
      loaded: false,
      shaded: true,
      wireframe: true,
      verticies: false,
    };
  }

  componentDidMount() {
    this.app = initScene(document.querySelector('#canvas'));
    this.app = initLights(this.app);
    this.cameraController = new CameraController();
    this.app = this.cameraController.initCamera(this.app);

    this.meshController = new MeshController();
    this.curveController = new CurvesController();

    fetch('/models/testBoat.json')
      .then(res => res.json())
      .then(data => {
        this.app = this.curveController.initCurves(this.app, data);
        this.curveController.updateFrames(this.app, data);
        this.meshController.initMesh(this.app, data);
        this.meshController.showMesh(this.app.displayShaded);
        // this.oldValues=JSON.parse(JSON.stringify(this.boatParametersService.updatePoint(data)));
        this.app.render();
        this.setState({
          loaded: true,
          boat: data,
        });
      });
  }

  handleDisplayChange = e => {
    if (this.state.loaded) {
      const { value } = e.target;
      const currentState = this.state[value];
      this.setState({
        [value]: !currentState,
      });
      if (value === 'shaded') {
        this.meshController.showMesh(!currentState);
      }
      if (value === 'verticies') {
        this.app.displayVerticies = !this.app.displayVerticies;

        Object.keys(this.state.boat).forEach(key => {
          if (key === 'width' || key === 'height' || key === 'length' || key === 'frames') {
            return;
          }
          this.app = this.curveController.deleteCurve(this.app, { key });
        });
        this.curveController.initCurves(this.app, this.state.boat);
      }
      if (value === 'wireframe') {
        this.app.displayWireFrame = !this.app.displayWireFrame;
        this.curveController.showCurves(!this.app.displayWireFrame);
      }
    }
  };

  render() {
    return (
      <div>
        <div id="canvas" className="canvas" />
        hei
        <div className="cube">
          <div className="camera-top" id="camera-top-button">
            Top
          </div>

          <div className="camera-side" id="camera-side-button">
            Side
          </div>

          <div className="camera-front" id="camera-front-button">
            Front
          </div>
        </div>
        <div className="camera-perspective" id="camera-45-button">
          45Â°
        </div>
        <div className="toggles-container">
          Toggle Shaded
          <input type="checkbox" value={this.state.shaded} onChange={this.handleDisplayChange} />
          Toggle Vertices
          <input type="checkbox" value={this.state.verticies} onChange={this.handleDisplayChange} />
          Toggle Wireframe
          <input type="checkbox" value={this.state.wireframe} onChange={this.handleDisplayChange} />
        </div>
        <div className="description">
          This was a project done for my senior project at New Mexico State University. The goal was
          to be able to construct a boat in 3D and output the blueprints so that an amateur boat
          maker could build it. The user would also be able to save and load their projects via json
          files as well as be able to export in obj or stl formats for 3D printing.
          <br />
          This demo allows you to view the demo boat and trigger some different rendering methods.
          You can also pan, zoom, and orbit around the boat.
        </div>
      </div>
    );
  }
}
