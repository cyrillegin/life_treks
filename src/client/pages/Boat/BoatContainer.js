import {compose, mapProps} from 'recompose';
import Boat from './Boat';

export default compose(
  mapProps((ownProps) => {
    return {
      ...ownProps,
      getBoat: (model) => {
        return fetch('/models/testBoat.json');
      },
    };
  }),
)(Boat);
