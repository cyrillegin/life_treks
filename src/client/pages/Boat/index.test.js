import React from 'react';
import renderer from 'react-test-renderer';
import Boat from './index';

describe.skip('Boat', () => {
  it('should render snapshots', () => {
    const tree = renderer.create(<Boat />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
