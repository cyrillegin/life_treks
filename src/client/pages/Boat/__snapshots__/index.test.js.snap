// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boat should render snapshots 1`] = `
<div>
  <div
    className="canvas"
    id="canvas"
  />
  hei
  <div
    className="cube"
  >
    <div
      className="camera-top"
      id="camera-top-button"
    >
      Top
    </div>
    <div
      className="camera-side"
      id="camera-side-button"
    >
      Side
    </div>
    <div
      className="camera-front"
      id="camera-front-button"
    >
      Front
    </div>
  </div>
  <div
    className="camera-perspective"
    id="camera-45-button"
  >
    45째
  </div>
  <div
    className="toggles-container"
  >
    Toggle Shaded
    <input
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
    Toggle Vertices
    <input
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
    Toggle Wireframe
    <input
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
  </div>
  <div
    className="description"
  >
    This was a project done for my senior project at New Mexico State University. The goal was to be able to construct a boat in 3D and output the blueprints so that an amateur boat maker could build it. The user would also be able to save and load their projects via json files as well as be able to export in obj or stl formats for 3D printing.
    <br />
    This demo allows you to view the demo boat and trigger some different rendering methods. You can also pan, zoom, and orbit around the boat.
  </div>
</div>
`;

exports[`Boat should test different displays 1`] = `"<div><div id=\\"canvas\\" class=\\"canvas\\"></div>hei<div class=\\"cube\\"><div class=\\"camera-top\\" id=\\"camera-top-button\\">Top</div><div class=\\"camera-side\\" id=\\"camera-side-button\\">Side</div><div class=\\"camera-front\\" id=\\"camera-front-button\\">Front</div></div><div class=\\"camera-perspective\\" id=\\"camera-45-button\\">45째</div><div class=\\"toggles-container\\">Toggle Shaded<input type=\\"checkbox\\" value=\\"false\\">Toggle Vertices<input type=\\"checkbox\\" value=\\"false\\">Toggle Wireframe<input type=\\"checkbox\\" value=\\"true\\"></div><div class=\\"description\\">This was a project done for my senior project at New Mexico State University. The goal was to be able to construct a boat in 3D and output the blueprints so that an amateur boat maker could build it. The user would also be able to save and load their projects via json files as well as be able to export in obj or stl formats for 3D printing.<br>This demo allows you to view the demo boat and trigger some different rendering methods. You can also pan, zoom, and orbit around the boat.</div></div>"`;

exports[`Boat should test different displays 2`] = `"<div><div id=\\"canvas\\" class=\\"canvas\\"></div>hei<div class=\\"cube\\"><div class=\\"camera-top\\" id=\\"camera-top-button\\">Top</div><div class=\\"camera-side\\" id=\\"camera-side-button\\">Side</div><div class=\\"camera-front\\" id=\\"camera-front-button\\">Front</div></div><div class=\\"camera-perspective\\" id=\\"camera-45-button\\">45째</div><div class=\\"toggles-container\\">Toggle Shaded<input type=\\"checkbox\\" value=\\"false\\">Toggle Vertices<input type=\\"checkbox\\" value=\\"true\\">Toggle Wireframe<input type=\\"checkbox\\" value=\\"true\\"></div><div class=\\"description\\">This was a project done for my senior project at New Mexico State University. The goal was to be able to construct a boat in 3D and output the blueprints so that an amateur boat maker could build it. The user would also be able to save and load their projects via json files as well as be able to export in obj or stl formats for 3D printing.<br>This demo allows you to view the demo boat and trigger some different rendering methods. You can also pan, zoom, and orbit around the boat.</div></div>"`;

exports[`Boat should test different displays 3`] = `"<div><div id=\\"canvas\\" class=\\"canvas\\"></div>hei<div class=\\"cube\\"><div class=\\"camera-top\\" id=\\"camera-top-button\\">Top</div><div class=\\"camera-side\\" id=\\"camera-side-button\\">Side</div><div class=\\"camera-front\\" id=\\"camera-front-button\\">Front</div></div><div class=\\"camera-perspective\\" id=\\"camera-45-button\\">45째</div><div class=\\"toggles-container\\">Toggle Shaded<input type=\\"checkbox\\" value=\\"false\\">Toggle Vertices<input type=\\"checkbox\\" value=\\"true\\">Toggle Wireframe<input type=\\"checkbox\\" value=\\"false\\"></div><div class=\\"description\\">This was a project done for my senior project at New Mexico State University. The goal was to be able to construct a boat in 3D and output the blueprints so that an amateur boat maker could build it. The user would also be able to save and load their projects via json files as well as be able to export in obj or stl formats for 3D printing.<br>This demo allows you to view the demo boat and trigger some different rendering methods. You can also pan, zoom, and orbit around the boat.</div></div>"`;
